src/volatile_memory.rs

```diff
@@ -1011,19 +1011,7 @@ impl<B: BitmapSlice> VolatileMemory for VolatileSlice<'_, B> {
     }
 
     fn get_slice(&self, offset: usize, count: usize) -> Result<VolatileSlice<B>> {
-        let _ = self.compute_end_offset(offset, count)?;
-        Ok(
-            // SAFETY: This is safe because the pointer is range-checked by compute_end_offset, and
-            // the lifetime is the same as self.
-            unsafe {
-                VolatileSlice::with_bitmap(
-                    self.addr.add(offset),
-                    count,
-                    self.bitmap.slice_at(offset),
-                    self.mmap,
-                )
-            },
-        )
```

Comment:
If that's the case, would it be better to use a `macro` to expand code here instead of calling one another 🤔

Reply:
I'd prefer keeping the duplicated code over a macro if we think that we don't want to have another layer of indirection here (which is a fair point, the indirection in this crate are already fairly complex as-is). But the compiler should be clever enough to figure out that it can inline the call to `subslice`, so no actual subroutine call should happen.